shigeifinder/1.3.2/Dockerfile

```diff
@@ -0,0 +1,139 @@
+ARG SHIGEIFINDER_VER="1.3.2"
+ARG SAMTOOLS_VER="1.10"
+ARG BWA_VER="0.7.17"
+
+FROM ubuntu:focal as app
+
+ARG SHIGEIFINDER_VER
+ARG SAMTOOLS_VER
+ARG BWA_VER
+
+LABEL base.image="ubuntu:focal"
+LABEL dockerfile.version="1"
+LABEL software="ShigEiFinder"
+LABEL software.version=${SHIGEIFINDER_VER}
+LABEL description="Cluster informed Shigella and EIEC serotyping tool from Illumina reads and assemblies"
+LABEL website="https://github.com/LanLab/ShigEiFinder"
+LABEL license="https://github.com/LanLab/ShigEiFinder/blob/main/LICENSE"
+LABEL maintainer1="Curtis Kapsak"
+LABEL maintainer1.email="curtis.kapsak@theiagen.com"
+
+# so that apt/tzdata doesn't ask for a timezone during build. Variable will not be in environment after building
+ARG DEBIAN_FRONTEND=noninteractive
+
+# install dependencies via apt; cleanup apt garbage
+# ncbi-blast+ from apt is v2.9.0 (min ver requirement for ShigEiFinder)
+# python version is 3.8.10
+RUN apt-get update && apt-get install --no-install-recommends -y \
+    make \
+    gcc \
+    g++ \
+    python3 \
+    python3-pip \
+    python3-setuptools \
+    zlib1g-dev \
+    wget \
+    ca-certificates \
+    procps \
+    libncurses5-dev \
+    libbz2-dev \
+    liblzma-dev \
+    libcurl4-gnutls-dev \
+    zlib1g-dev \
+    libssl-dev \
+    bzip2 \
+    gawk \
+    gnuplot \
+    ncbi-blast+ \
+    unzip && \
+    rm -rf /var/lib/apt/lists/* && apt-get autoclean
+
+# install bwa
+RUN mkdir /bwa && \
+    cd /bwa && \
+    wget https://github.com/lh3/bwa/releases/download/v${BWA_VER}/bwa-${BWA_VER}.tar.bz2 && \
+    tar -xjf bwa-${BWA_VER}.tar.bz2 && \
+    rm bwa-${BWA_VER}.tar.bz2 && \
+    cd bwa-${BWA_VER} && \
+    make
+
+# install samtools; no need to add to PATH, 'make install' does this for us
+RUN wget https://github.com/samtools/samtools/releases/download/${SAMTOOLS_VER}/samtools-${SAMTOOLS_VER}.tar.bz2 && \
+    tar -xjf samtools-${SAMTOOLS_VER}.tar.bz2 && \
+    rm samtools-${SAMTOOLS_VER}.tar.bz2 && \
+    cd samtools-${SAMTOOLS_VER} && \
+    ./configure && \
+    make && \
+    make install
+
+# install ShigEiFinder; make /data
+RUN wget https://github.com/LanLab/ShigEiFinder/archive/refs/tags/v${SHIGEIFINDER_VER}.tar.gz && \
+    tar -xvf v${SHIGEIFINDER_VER}.tar.gz && \
+    rm v${SHIGEIFINDER_VER}.tar.gz && \
+    cd ShigEiFinder-${SHIGEIFINDER_VER} && \
+    python3 setup.py install && \
+    mkdir /data
+
+# final working directory is /data
+WORKDIR /data
+
+# set locale settings for singularity compatibility. Set PATH to include bwa
+ENV LC_ALL=C \
+    PATH="${PATH}:/bwa/bwa-${BWA_VER}"
+
+# test layer
+FROM app as test
+
+ARG SHIGEIFINDER_VER
+
+# install ncbi datasets tool (pre-compiled binary); place in $PATH
+RUN wget https://ftp.ncbi.nlm.nih.gov/pub/datasets/command-line/LATEST/linux-amd64/datasets && \
+    chmod +x datasets && \
+    mv -v datasets /usr/local/bin
```

Comment:
In the test, this gives a warning message:
```
#13 2.021 New version of client (14.0.1) available at https://ftp.ncbi.nlm.nih.gov/pub/datasets/command-line/LATEST/linux-amd64/datasets
```

Which I don't think you can do anything about.

Reply:
yeah...we should be pulling the latest version given the FTP URL 🤷